From 5e01b16282245168a23ec981a81468cb310048d8 Mon Sep 17 00:00:00 2001
From: fanzha3x <fanx.zhang@intel.com>
Date: Thu, 28 Jan 2021 10:11:25 +0800
Subject: [PATCH] Fix accessing SD card error for recording

API Environment.getExternalStorageDirectory is obsolete,
replace with context.getExternalFilesDir.
Access App-specific directory does not need any permissions.

Tracked-On: ACP-1473
Change-Id: I90756107bfa5654230ebb42846dc0651fba31a3e
Signed-off-by: fanzha3x <fanx.zhang@intel.com>
---
 src/com/android/soundrecorder/Recorder.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/soundrecorder/Recorder.java b/src/com/android/soundrecorder/Recorder.java
index 9d906c2..bc01cec 100644
--- a/src/com/android/soundrecorder/Recorder.java
+++ b/src/com/android/soundrecorder/Recorder.java
@@ -147,7 +147,7 @@ public class Recorder implements OnCompletionListener, OnErrorListener {
         stop();
         
         if (mSampleFile == null) {
-            File sampleDir = Environment.getExternalStorageDirectory();
+            File sampleDir = context.getExternalFilesDir(null);
             if (!sampleDir.canWrite()) // Workaround for broken sdcard support on the device.
                 sampleDir = new File("/sdcard/sdcard");
             
-- 
2.25.1

